docs/concepts/CLQL.md

```diff
@@ -253,31 +253,231 @@ Results under include statements appear as children of the parent of the corresp
+
+The following query finds function calls at the top level of a file and follows the `calls` edge to their definition:
+
+```
+common.func_call:
+  edge("calls"):
+    common.func
+```
+
+<br />
+
+# Path
+
+Path statements encapsulate CLQL trees. These subtrees can be repeated with a single argument allowing succint repition of complex patterns. Branched paths can rejoin allowing a fact to match nodes with different kinds of parents.
+
+## Linear
+
+Say we wanted to find triply nested if statements, our query would look like the following:
+
+```clql
+common.if_stmt:
+  common.if_stmt:
+    common.if_stmt
+```
+
+With paths, we can express the same thing like so:
+
+```clql
+path(repeat = 3):
+  common.if_stmt:
+    pathcontinue
+```
+
+Once a query reaches a `pathcontinue` statement it continues from the `path` statement until the path has been repeated the specified number of times.
+
+## Repeat range
+
+Some queries cannot be written with `path` statements. Say we wanted to find all functions called by `someFunc()` and an arbitrarily long chain of calls. Our query would have to explicitly match either directly called functions, or functions with 1, 2, 3 etc intermediaries to infinity.
+
+```clql
+common.func:
+  name == "someFunc"
+  any_of:
+    common.func_call(depth = any):
+      edge("calls"):
+        common.func
+    common.func_call(depth = any):
+      edge("calls"):
+        common.func:
+          common.func_call(depth = any):
+            edge("calls"):
+              common.func
+    ...
+    common.func_call(depth = any):
+      edge("calls"):
+        common.func:
+          common.func_call(depth = any):
+            edge("calls"):
+              common.func:
+                ...
+```
+
+With paths the same query is trivial:
+
+```clql
+common.func:
+  name == "someFunc"
+  path(repeat = 1:):
+    common.func_call(depth = any):
+      edge("calls"):
+        common.func:
+          pathcontinue
+```
+
+`repeat = 1:` is a range specifying that the path should be repeated one or more times.
+
+## Complex subtrees
+
+Say we wanted to match triply nested if statements that all check the same value, our query would look like the following:
+
+```clql
+common.if_stmt:
+  common.condition:
+    common.var:
+      name as varName
+  common.if_stmt:
+    common.condition:
+      common.var:
+        name == varName
+    common.if_stmt:
+      common.condition:
+        common.var:
+          name == varName
+```
+
+With paths our query has much less repitition:
+
+```clql
+common.func:
+  path(repeat = 3):
+    common.if_stmt:
+      common.condition:
+        common.var:
+          name as varName
+      pathcontinue
+```
+
+Note that CLQL elements that are children of `path`, not just the `if_stmt`. Also note that repeated definitions of `varName` are replaced with assertions.
```

Comment:
We need more subtlety around vars and paths. 

The key use case (for PHP SQL injection) is using and defining a variable within a block to create distinct vars for each repetition. There will be a use case for repeating the same assertion, but not defining anything. The above is also a valid use case.